python/foglamp/services/core/api/common.py

```diff
@@ -99,9 +99,10 @@ def get_stats(k):
         raise web.HTTPException(reason=str(ex))
 
 
-def do_shutdown(loop=None):
+def do_shutdown(request):
     _logger.info("Executing controlled shutdown")
-    if loop is None:
-        loop = asyncio.get_event_loop()
-    loop.run_until_complete(server.Server._stop())
-    loop.stop()
+    try:
+        loop = request.loop
+        loop.run_until_complete(server.Server.shutdown(request))
```

Comment:
we should fix `shutdown` too, To use request.loop instead of `loop = asyncio.get_event_loop()` and in for async sleep there use same loop `await asyncio.sleep(2.0, loop= <request.loop>)`